File: regalia-logansquare/src/test/java/com/commit451/regalia/logansquare/Cat.java
package com.commit451.regalia.logansquare;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * A cat!
 */
@JsonObject
public class Cat extends RealmObject {

    @PrimaryKey
    @JsonField(name = "name")
    String name;
    @JsonField(name = "friends", typeConverter = CatListConverter.class)
    RealmList<Cat> friends;
}

File: regalia-logansquare/src/test/java/com/commit451/regalia/logansquare/Cat.kt
package com.commit451.regalia.logansquare

import com.bluelinelabs.logansquare.annotation.JsonField
import com.bluelinelabs.logansquare.annotation.JsonObject
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

/**
 * A cat!
 */
@JsonObject
open class Cat : RealmObject() {
    @PrimaryKey
    @JsonField(name = ["name"])
    var name: String? = null
    @JvmField
    @JsonField(name = ["friends"], typeConverter = CatListConverter::class)
    var friends: RealmList<Cat>? = null
}